Error handling, logging, and string manipulation.

paulp opened the pull request.

paulp committed:
Rename Lines to Strings.

Lines is just a wrapper around a string slice, and isn't
concerned with the meaning of those strings.

paulp committed:
Error handling, logging, and string manipulation.

Eliminates logging dependency. Logs more consistently.

Introduces docopt for command line option parsing.

More consistent error handling.

EECOLOR commented on suffuse/opts.go:
```diff
@@ -1,127 +1,107 @@
-    // Parse has already printed the problem and usage, there is no way to tell it not to do that
-    return nil, &SfsConfigError{err: err.Error(), usage: nil}
+func optsFromArgs(args []string) *sfsOpts {
+  argMap, err := docopt.Parse(sfsUsage(), args, /*help*/true, /*version*/"HEAD", /*optionsFirst*/false, /*exit*/false)
```
Thank you for putting those comments there

paulp replied:
Yeah, I have no idea how people come up with such garbage APIs at this stage.

EECOLOR commented on suffuse/opts.go:
```diff
@@ -1,127 +1,107 @@
+  -m PATH     mount point
+  -n VOLNAME  volume name (OS X only)
+  -t          create scratch directory as mount point
+  -v          more verbose logging (also -vv and -vvv)
```
So I guess that library turns `-v` into `1` and `-vv` into `2`?

paulp replied:
Right.

EECOLOR replied:
Never mind, figured it out. Btw, that library is pretty amazing

EECOLOR committed:
Moved around some code, fixed some missed renames from `Lines` to `Strings` and simplified `Strings` type

- Moved `NoError` to `error.go`
- Moved `OsStderr` to `process.go`
- Introduced some spaces
- Renamed `lines_test.go` to `strings_test.go`
- Removed `SfsConfigError`
- Added default log level in case someone goes nuts on the `v`
- Moved `Names` constructor closer to `Name`
- Moved some (less important) functions down in files
- Gave `Regex` its own file
- Changed `Strings` from a struct to an alias of `[]string`
- Removed `Strs(...)` function in favor of `Strings{...}`
- Added `Array()` method on `Strings` to replace the `strings` property

EECOLOR commented:
Suggested changes in new commit
